tests/unit/vault_test.cairo

```diff
@@ -1,150 +1,413 @@
+use core::debug;
+use openzeppelin::token::erc20::interface::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait};
+use openzeppelin::token::erc721::interface::{ERC721ABIDispatcher, ERC721ABIDispatcherTrait};
+use pwn::interfaces::pool_adapter::{IPoolAdapterDispatcher, IPoolAdapterDispatcherTrait};
+use pwn::mocks::mock_erc20::MockERC20;
+use pwn::mocks::mock_erc721::MockERC721;
+use pwn::mocks::mock_pool_adapter::MockPoolAdapter;
+use pwn::mocks::mock_vault::PwnVaultTestContract;
+use pwn::mocks::mock_vault::{
+    IMockPwnVaultTestContractDispatcher, IMockPwnVaultTestContractDispatcherTrait
+};
+
+use snforge_std::{
+    declare, ContractClassTrait, store, load, map_entry_address, start_cheat_caller_address,
+    CheatSpan, cheat_caller_address, spy_events, mock_call, EventSpy, EventSpyTrait,
+    EventSpyAssertionsTrait
+};
+use starknet::ContractAddress;
+
+#[starknet::interface]
+trait IERC20Mock<TState> {
+    fn mint(ref self: TState, recipient: ContractAddress, amount: u256);
+}
+
+#[starknet::interface]
+trait IERC7210Mock<TState> {
+    fn mint(ref self: TState, recipient: ContractAddress, tokenId: u256);
+}
+
+fn ALICE() -> ContractAddress {
+    starknet::contract_address_const::<'Alice'>()
+}
+
+fn BOB() -> ContractAddress {
+    starknet::contract_address_const::<'Bob'>()
+}
+
+fn deploy() -> (
```

Comment:
you should return dispatchers directly so you don't have to instantiate one every time see [here](https://github.com/NethermindEth/pwn-starknet/blob/54d60a8265f3f39e40687533ebaa8d5f90a1f610/tests/unit/multitoken_library_test.cairo#L29)